cns/restserver/internalapi.go

```diff
@@ -325,6 +342,24 @@ func (service *HTTPRestService) ReconcileNCState(ncRequest *cns.CreateNetworkCon
 	return 0
 }
 
+func podInfoByIPToPodInfoExtByInterfaceID(podInfoByIP map[string]cns.PodInfo) map[string]podInfoExt {
+	out := make(map[string]podInfoExt)
+	for ip, podInfo := range podInfoByIP {
+		k := podInfo.InterfaceID()
+
+		xt := out[k]
+		xt.PodInfo = podInfo
+		xt.IPs = append(xt.IPs, ip)
+		out[k] = xt
+	}
+	return out
+}
+
+type podInfoExt struct {
```

Comment:
podIPInfo?

Reply:
Pod "IP" info feels a little incomplete because it holds multiple IPs. I wanted to just extend `cns.PodInfo` at first but that turns out to be a much larger change (not sure why it's an interface, it has a single implementation as far as I could see). This is just a small type that extends `cns.PodInfo`, thus the suffix.